core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/dr/HandlerMethods.java

```diff
@@ -0,0 +1,39 @@
+package com.sequenceiq.cloudbreak.reactor.handler.cluster.dr;
+
+import com.sequenceiq.cloudbreak.orchestrator.model.SaltConfig;
+import com.sequenceiq.cloudbreak.orchestrator.model.SaltPillarProperties;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.util.Collections.singletonMap;
+
+public class HandlerMethods {
+
+    private HandlerMethods() {
+
+    }
+
+    public static SaltConfig createSaltConfig(String location, String backupId, String cloudPlatform) {
```

Comment:
I'm of the opinion that correctly handling paths is scary/hard, but this also decently easy to test.

Could you run some unit tests against the `scheme` or `fullLocation` handling?

Refactoring to something like:
```java
public static SaltConfig createSaltConfig(String location, String backupId, String cloudPlatform) {
        String fullLocation = buildFullLocation(location, backupId, cloudPlatform); // test me

        Map<String, SaltPillarProperties> servicePillar = new HashMap<>();
        servicePillar.put("disaster-recovery", new SaltPillarProperties("/postgresql/disaster_recovery.sls",
            singletonMap("disaster_recovery", singletonMap("object_storage_url", fullLocation))));
        return new SaltConfig(servicePillar);
    }

public static String buildFullLocation (String location, String backupId, String cloudPlatform) {
        String fullLocation = location + "/" + backupId + "_database_backup";
        String scheme;
        if ("aws".equalsIgnoreCase(cloudPlatform)) {
            scheme = "s3:/";
        } else if ("azure".equalsIgnoreCase(cloudPlatform)) {
            // TODO verify this is right when azure flow is working
            scheme = "abfs:/";
        } else {
            throw new UnsupportedOperationException("Cloud platform " + cloudPlatform + "not supported for backup/restore");
        }

        if (fullLocation.startsWith("/")) {
            fullLocation = scheme + fullLocation;
        } else {
            fullLocation = scheme + fullLocation.substring(fullLocation.indexOf("//") + 1);
        }
        return fullLocation;
}

Reply:
I can write some unit tests around that, but am I handling paths incorrectly here? Outside of the requested refactor?

Reply:
I don't think we handle a relative path like `bucket/backups` correctly here:
```java
fullLocation = scheme + fullLocation.substring(fullLocation.indexOf("//") + 1);
```
We'd get `s3:/bucket/backups`, right? I'm not sure how resilient this method needs to be.

To be pedantic, we may want to define the schemes without the slash to follow the URI specification, if we ever do try to use them with the [URI constructor](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#URI-java.lang.String-java.lang.String-java.lang.String-) I think it would blow up.

I'd consider using the `URI` class to at least guarantee that we have a valid URI coming out of this, though it may not be a valid s3/abfs/hdfs uri.

Reply:
I was using the URI class originally, but it wasn't doing exactly what I wanted so I changed it. That's no excuse for not digging into what I was doing wrong though :). I've updated the code to use URIs and added unit tests around it.